src/theory/bv/int_blaster.cpp

```diff
@@ -194,8 +215,352 @@ Node IntBlaster::translateWithChildren(
     const std::vector<Node>& translated_children,
     std::vector<Node>& lemmas)
 {
-  Node binarized = makeBinary(original);
-  // continue to process the binarized version
+  // The translation of the original node is determined by the kind of
+  // the node.
+  kind::Kind_t oldKind = original.getKind();
+
+  // Some BV operators were eliminated before this point.
+  Assert(oldKind != kind::BITVECTOR_SDIV);
+  Assert(oldKind != kind::BITVECTOR_SREM);
+  Assert(oldKind != kind::BITVECTOR_SMOD);
+  Assert(oldKind != kind::BITVECTOR_XNOR);
+  Assert(oldKind != kind::BITVECTOR_NAND);
+  Assert(oldKind != kind::BITVECTOR_SUB);
+  Assert(oldKind != kind::BITVECTOR_REPEAT);
+  Assert(oldKind != kind::BITVECTOR_ROTATE_RIGHT);
+  Assert(oldKind != kind::BITVECTOR_ROTATE_LEFT);
+  Assert(oldKind != kind::BITVECTOR_COMP);
+  Assert(oldKind != kind::BITVECTOR_SGT);
+  Assert(oldKind != kind::BITVECTOR_SLE);
+  Assert(oldKind != kind::BITVECTOR_SGE);
+  Assert(oldKind != kind::EXISTS);
+
+  // BV division by zero was eliminated before this point.
+  Assert(oldKind != kind::BITVECTOR_UDIV
+         || !(original[1].isConst()
+              && original[1].getConst<BitVector>().getValue().isZero()));
+
+  // Store the translated node
+  Node returnNode;
+
+  // Translate according to the kind of the original node.
+  switch (oldKind)
+  {
+    case kind::BITVECTOR_ADD:
+    {
+      Assert(original.getNumChildren() == 2);
+      uint64_t bvsize = original[0].getType().getBitVectorSize();
+      std::cout << "panda before" << std::endl;
```

Comment:
Ditto, remove.

Reply:
sorry for overlooking these debug messages, removed. in 14ac299.